api_router/src/PokemonDetail.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './PokemonDetail.css';
+
+export default class PokemonDetail extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      pokemonData: {},
+      imgUrl: ""
+    };
+    this.getStats = this.getStats;
+  }
+
+  static propTypes = {
+    name: PropTypes.string,
+    url: PropTypes.string
+  }
+
+  getStats() {
+    if (Object.keys(this.state.pokemonData).length > 0) {
+      return (
+        <ul className="pokeStats">
+          <li>{this.state.pokemonData.stats[0].stat.name}: {this.state.pokemonData.stats[0].base_stat}</li>
+          <li>{this.state.pokemonData.stats[1].stat.name}: {this.state.pokemonData.stats[1].base_stat}</li>
+          <li>{this.state.pokemonData.stats[2].stat.name}: {this.state.pokemonData.stats[2].base_stat}</li>
+          <li>{this.state.pokemonData.stats[3].stat.name}: {this.state.pokemonData.stats[3].base_stat}</li>
+          <li>{this.state.pokemonData.stats[4].stat.name}: {this.state.pokemonData.stats[4].base_stat}</li>
+        </ul>
+      )
+    } else {
+      return "just one sec..."
+    }
+  }
+
+  componentDidMount() {
+    fetch(this.props.url)
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        this.setState({
+          pokemonData: data,
+          imgUrl: data.sprites.front_default
+        })
+        this.dataReady=true;
```

Comment:
Just a reminder that instance variables won't cause any logic to rerun in the render function 😄. But it looks like you already figured that out by checking if `pokemonData` has values.